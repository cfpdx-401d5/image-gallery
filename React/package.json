{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
