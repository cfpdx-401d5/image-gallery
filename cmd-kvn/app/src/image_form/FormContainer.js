import React, { Component, PropTypes } from 'react';

import InputField from './InputField';

class FormContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      description: '',
      url: '',
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(e) {
    const target = e.target;
    const value = target.value;
    const name = target.name;
    console.log(e.target.name);

    this.setState({
      [name]: value 
    });
  }

  render() {
    return (
        <form onSubmit={(e) => {
          e.preventDefault();
          this.props.addImage(this.state);
          e.target.reset();
          this.setState({ title: '', description: '', url: '' });
        }}
        >
          <h4>Add a new image</h4>
          <InputField name='title' handler={this.handleInputChange} />
          <InputField name='description' handler={this.handleInputChange} />
          <InputField name='url' handler={this.handleInputChange} />
          <button type='submit'>Submit</button>
        </form>
    );
  }
}

FormContainer.propTypes = {
  addImage: PropTypes.func.isRequired,
};

export default FormContainer;
