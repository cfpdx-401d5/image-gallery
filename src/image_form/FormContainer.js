import React, { Component, PropTypes } from 'react';
import shortid from 'shortid';

import InputField from './InputField';

class FormContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      description: '',
      url: '',
      id: '',
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(e) {
    const target = e.target;
    const value = target.value;
    const name = target.name;
    console.log(e.target.name);

    this.setState({
      [name]: value 
    });
  }

  render() {
    return (
        <form onSubmit={(e) => {
          this.setState({ id: shortid.generate() });
          this.props.addImage(e, this.state);
          e.target.reset();
        }}
        >
          <h4>Add a new image</h4>
          <InputField name='title' handler={this.handleInputChange} />
          <InputField name='description' handler={this.handleInputChange} />
          <InputField name='url' handler={this.handleInputChange} />
          <button type='text'>Submit</button>
        </form>
    );
  }
}

FormContainer.propTypes = {
  addImage: PropTypes.func.isRequired,
};

export default FormContainer;
