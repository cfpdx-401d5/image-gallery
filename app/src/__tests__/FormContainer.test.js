/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FormContainer from '../image_form/FormContainer';

describe('FormContainer', function () {
    it('runs the FormContainer', function () {
        const rendered = shallow(<FormContainer addImage={() => {}} />
);
        expect(toJson(rendered)).toMatchSnapshot();
    })
});