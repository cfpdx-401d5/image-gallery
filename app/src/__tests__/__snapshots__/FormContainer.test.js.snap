exports[`FormContainer runs the FormContainer 1`] = `
<form
  onSubmit={[Function]}>
  <h4>
    Add a new image
  </h4>
  <InputField
    handler={[Function]}
    name="title" />
  <InputField
    handler={[Function]}
    name="description" />
  <InputField
    handler={[Function]}
    name="url" />
  <button
    type="text">
    Submit
  </button>
</form>
`;
